diag_manager/fms_diag_yaml.F90

```diff
@@ -983,9 +996,7 @@ end function has_file_write
 !! @return true if obj%file_sub_region sub region variables are allocated
 pure logical function has_file_sub_region (obj)
   class(diagYamlFiles_type), intent(in) :: obj !< diagYamlFiles_type object to initialize
-  if ( (allocated(obj%file_sub_region%grid_type) .and. allocated(obj%file_sub_region%lat_lon_sub_region)) &
-   .or.(allocated(obj%file_sub_region%grid_type) .and. allocated(obj%file_sub_region%index_sub_region)))  &
-   then
+  if ( trim(obj%file_sub_region%grid_type) .eq. "latlon" .or. trim(obj%file_sub_region%grid_type) .eq. "index") then
```

Comment:
These should be parameter variables instead of hard coded.  Imagine having to change this down the road...